app/models/git_repository.rb

```diff
@@ -114,6 +114,10 @@ def update_repo
     with_worktree do |worktree|
       message = "Updating #{url} in #{directory_name}..."
       _log.info(message)
+      # If the remote url has changed set it to the new url
+      git_transaction { worktree.url = url } if worktree.url != url
+
+      # Fetch and rebase the git worktree
       worktree.send(:pull)
```

Comment:
Unrelated, but any idea why `GitWorktree#pull` is private?  A number of these methods look like they would be useful (e.g. `fetch`, `merge`, `commit`, `rebase`)

Reply:
I'm not 100% sure, but there are some weird locking issues around specific methods (I think pull is one of them), so some of those might have been lower level methods that were intentionally hidden.  `pull` for example, is a fetch and merge, and merge is creating a merge commit then moving the branch label.  Since there are multiple operations they need to be done atomically. otherwise 2 separate threads could cleave into each other and you end up with a broken history.

Reply:
`#pull` does have a lock around it https://github.com/ManageIQ/manageiq/blob/master/lib/git_worktree.rb#L331-L333 but not sure why that means it would have to be private.  I could see the individual non-locked methods being private but pull appears to be safe to call as a public method (otherwise we'd be in trouble here)

Reply:
I'll open another PR and we can discuss there

Reply:
Yes, I was talking about individual non-locked methods being private...it if it locked, I see no reason it can't be public.